listenbrainz/webserver/rabbitmq_connection.py

```diff
@@ -7,28 +7,27 @@
 _rabbitmq = None
 
 def init_rabbitmq_connection(app):
-    """Create a connection to the RabbitMQ server."""
+    """Initialize the webserver rabbitmq connection.
+
+    This initializes _rabbitmq as a connection pool from which new RabbitMQ
+    connections can be acquired.
+    """
     global _rabbitmq
 
     if "RABBITMQ_HOST" not in app.config:
         app.logger.error("RabbitMQ host:port not defined. Sleeping 2 seconds, and exiting.")
         sleep(2)
         sys.exit(-1)
 
-    connection_config = {
-        'username': app.config['RABBITMQ_USERNAME'],
-        'password': app.config['RABBITMQ_PASSWORD'],
-        'host': app.config['RABBITMQ_HOST'],
-        'port': app.config['RABBITMQ_PORT'],
-        'virtual_host': app.config['RABBITMQ_VHOST']
-    }
-
-    connection = utils.connect_to_rabbitmq(**connection_config,
-                                           error_logger=app.logger.error,
-                                           error_retry_delay=2)
+    connection_parameters = pika.ConnectionParameters(
+        host=app.config['RABBITMQ_HOST'],
+        port=app.config['RABBITMQ_PORT'],
+        virtual_host=app.config['RABBITMQ_VHOST'],
+        credentials=pika.PlainCredentials(app.config['RABBITMQ_USERNAME'], app.config['RABBITMQ_PASSWORD']),
+    )
 
     _rabbitmq = pika_pool.QueuedPool(
-        create=lambda: connection,
+        create=lambda: pika.BlockingConnection(connection_parameters),
```

Comment:
There is no need (or value) of looping here because connections are acquired here (https://github.com/metabrainz/listenbrainz-server/blob/master/listenbrainz/webserver/views/api_tools.py#L72), so the loop is unneccessary.